src/java/picard/analysis/directed/InsertSizeMetricsCollector.java

```diff
@@ -113,7 +115,17 @@ public double getTotalInserts() {
         }
 
         public void addMetricsToFile(final MetricsFile<InsertSizeMetrics,Integer> file) {
-            for (final Histogram<Integer> h : this.Histograms.values()) totalInserts += h.getCount();
+            // get the number of inserts, and the maximum and minimum keys across, across all orientations
+            int minKey = Integer.MAX_VALUE, maxKey = Integer.MIN_VALUE;
+            for (final Histogram<Integer> h : this.Histograms.values()) {
+                totalInserts += h.getCount();
+                if (!h.isEmpty()) {
+                    int curMin = Collections.min(h.keySet()), curMax = Collections.max(h.keySet());
```

Comment:
I think this would be clearer written as:
```java
minKey = Math.min(minKey, Collections.min(h.keySet())
maxKey = Math.max(maxKey, Collections.max(h.keySet())
```